### PR TITLE
Additional synchronization fixes (inclusive of the last set) from sync validation testing

### PR DESCRIPTION
**Pull request by jzulauf-lunarg**
Disabling FORCE_FULL_BARRIER exposed additional synchronization issues (mostly missing and/or incorrect stages.  Some attempts at creating logical, minimal sets of stage and access masks is attempted, but the author is unfamiliar enough with the codebase to suggest that this set is complete/appropriate/most-efficient.

Note that an outstanding hazard between sdfgi compute buffer update results from an invalid use of buffer_update within a compute list and is not addressed here.

Testing was against the "sponza" scene file.

### CODE REVIEW

**Comment by jzulauf-lunarg** [comment]
@reduz  -- for you, my friend.

File: drivers/vulkan/rendering_device_vulkan.cpp [hunk 1]
```diff
@@ -2512,6 +2536,8 @@ Vector<uint8_t> RenderingDeviceVulkan::texture_get_data(RID p_texture, uint32_t
 			image_memory_barrier.pNext = nullptr;
 			image_memory_barrier.srcAccessMask = VK_ACCESS_TRANSFER_READ_BIT;
 			image_memory_barrier.dstAccessMask = VK_ACCESS_SHADER_READ_BIT;
+			if (tex->usage_flags & TEXTURE_USAGE_STORAGE_BIT)
+				image_memory_barrier.dstAccessMask |= VK_ACCESS_SHADER_WRITE_BIT;
```
Comment (akien-mga) [inline drivers/vulkan/rendering_device_vulkan.cpp:2540]:
```suggestion
			if (tex->usage_flags & TEXTURE_USAGE_STORAGE_BIT) {
				image_memory_barrier.dstAccessMask |= VK_ACCESS_SHADER_WRITE_BIT;
			}
```

We're now enforcing this style, but `clang-format` can't do it for you sadly :|

File: drivers/vulkan/rendering_device_vulkan.cpp [hunk 2]
```diff
@@ -2485,6 +2536,9 @@ Vector<uint8_t> RenderingDeviceVulkan::texture_get_data(RID p_texture, uint32_t
 			image_memory_barrier.pNext = nullptr;
 			image_memory_barrier.srcAccessMask = VK_ACCESS_TRANSFER_READ_BIT;
 			image_memory_barrier.dstAccessMask = VK_ACCESS_SHADER_READ_BIT;
+			if (tex->usage_flags & TEXTURE_USAGE_STORAGE_BIT) {
```
Comment (jzulauf-lunarg) [inline drivers/vulkan/rendering_device_vulkan.cpp:2539]: @akien-mga  -- okay then?

File: drivers/vulkan/rendering_device_vulkan.cpp (hunk 1 above)
Comment (jzulauf-lunarg) [reply drivers/vulkan/rendering_device_vulkan.cpp]: updated

File: drivers/vulkan/rendering_device_vulkan.cpp (hunk 2 above)
Comment (akien-mga) [reply drivers/vulkan/rendering_device_vulkan.cpp:2539]: Yes, thanks :)

**Comment by jzulauf-lunarg** [comment]
replaced by newer more complete version.

**Pull request closed without merging**